list.go

```diff
@@ -17,24 +17,132 @@ package hessian
 import (
 	"io"
 	"reflect"
+	"strings"
+	"time"
 )
 
 import (
 	perrors "github.com/pkg/errors"
 )
 
+var (
+	listTypeNameMapper = map[string]string{
+		"string": "[string",
+
+		"int8": "[short",
+
+		"int16":  "[short",
+		"uint16": "[short",
+
+		"int32":  "[int",
+		"uint32": "[int",
+
+		"int":  "[long",
+		"uint": "[long",
+
+		"int64":  "[long",
+		"uint64": "[long",
+
+		"float32": "[float",
+		"float64": "[double",
+
+		"bool": "[boolean",
+
+		"time.Time": "[date",
+	}
+	listTypeMapper = map[string]reflect.Type{
+		"[string":           reflect.TypeOf([]string{""}),
+		"[java.lang.String": reflect.TypeOf([]string{""}),
+		"[char":             reflect.TypeOf([]string{""}),
+		"[short":            reflect.TypeOf([]int32{0}),
+		"[int":              reflect.TypeOf([]int32{0}),
+		"[long":             reflect.TypeOf([]int64{0}),
+		"[float":            reflect.TypeOf([]float64{0}),
+		"[double":           reflect.TypeOf([]float64{0}),
+		"[boolean":          reflect.TypeOf([]bool{true}),
+		"[java.util.Date":   reflect.TypeOf([]time.Time{{}}),
+		"[date":             reflect.TypeOf([]time.Time{{}}),
+	}
+	noType = perrors.New("no type name.")
+)
+
+func registerTypeName(gotype, javatype string) {
+	listTypeNameMapper[gotype] = "[" + javatype
+}
+func getType(javalistname string) reflect.Type {
+	sliceTy := listTypeMapper[javalistname]
+	if sliceTy == nil {
+		javaname := strings.Trim(javalistname, "[")
+		tpStructInfo, _ := getStructInfo(javaname)
+		tp := tpStructInfo.typ
+		if tp == nil {
+			return nil
+		}
+		if tp.Kind() != reflect.Ptr {
+			tp = reflect.New(tp).Type()
+		}
+		sliceTy = reflect.SliceOf(tp)
+	}
+	return sliceTy
+}
+
 /////////////////////////////////////////
 // List
 /////////////////////////////////////////
 
-// # list/vector
+// encList write list
+func (e *Encoder) encList(v interface{}) error {
```

Comment:
Why not just write `encList` like below

```go
func (e *Encoder) encList(v interface{}) error {
    if reflect.TypeOf(v).String() != "[]interface {}" {
        return e.writeTypedList(v)
    } else {
        return e.writeUntypedList(v)
   }
}
```

I prefer to raising `notype` error directly.

Reply:
as above